### PR TITLE
Enabling the host application to control the timing of attaching the |FlutterView| to the engine

### PR DESCRIPTION
**Pull request by 0xZOne**
In the add-to-app scenario where multiple FlutterViews share the same FlutterEngine, the host application desires to determine the timing of attaching the FlutterView to the engine, for example, during the `onResume` instead of the `onCreateView`.

As an example, consider the following scenario: A native page contains multiple tabs, and each tab is a FlutterFragment. During initialization, FlutterFragments of different tabs are created almost simultaneously, but only the one that needs to be displayed currently requires attachment to the engine, while the others need to be attached only when they receive the |onResume| callback.

Partial fix: https://github.com/flutter/flutter/issues/130235

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by chinmaygarde** [comment]
cc @reidbaker to find a reviewer.

**Comment by 0xZOne** [comment]
@johnmccutchan Please take a look at it when you have a chance. Many thanks.

**Comment by chinmaygarde** [comment]
Triage: The comments on the linked issue seem to indicate that underlying issue had a workaround. Is this PR stale?

**Comment by 0xZOne** [comment]
> Triage: The comments on the linked issue seem to indicate that underlying issue had a workaround. Is this PR stale?

No, it's not stale. I'm still waiting on a code review.

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java [hunk 1]
```diff
@@ -1407,6 +1407,19 @@ public boolean shouldDispatchAppLifecycleState() {
...
+  /**
+   * Whether to manually attach the {@link FlutterView} to the engine by the host application.
+   *
+   * <p>Returning {@code true} means that the responsibility of attaching the {@link FlutterView} to
+   * the engine will be taken over by the host application.
+   *
+   * <p>Defaults to {@code false}.
+   */
+  @Override
+  public boolean shouldAttachToEngineManually() {
```
[the remark this answers is not in the document]
Comment (0xZOne) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivity.java]: done.

**Review by johnmccutchan — APPROVED** [review]

**Pull request merged**